pkgs/tools/misc/pastebinit/default.nix

```diff
@@ -3,47 +3,74 @@
 , fetchurl
 , fetchpatch
 , python3
+, libxslt
+, docbook-xsl-ns
+, installShellFiles
+, wget
+, cacert
 }:
-stdenv.mkDerivation rec {
-  version = "1.5";
+stdenv.mkDerivation
+rec {
+  version = "1.6.2-1";
   pname = "pastebinit";
 
   src = fetchurl {
-    url = "https://launchpad.net/pastebinit/trunk/${version}/+download/${pname}-${version}.tar.bz2";
-    sha256 = "0mw48fgm9lyh9d3pw997fccmglzsjccf2y347gxjas74wx6aira2";
+    url = "https://launchpad.net/ubuntu/+archive/primary/+sourcefiles/${pname}/${version}/${pname}_${lib.removeSuffix "-1" version}.orig.tar.gz";
+    sha256 = "b7f63207bfb66eca0e77052283815346d3ad9de2b30aa9f68b74521f0b54940a";
   };
 
-  buildInputs = [
-    (python3.withPackages (p: [ p.distro ]))
+  patches = [
+    ./use-drv-etc.patch
   ];
 
-  patchFlags = [ "-p0" ];
+  nativeBuildInputs = [
+    libxslt
+    installShellFiles
+  ];
 
-  patches = [
-    # Required to allow pastebinit 1.5 to run on Python 3.8
-    ./use-distro-module.patch
-    # Required to remove the deprecation warning of FancyURLopener
-    ./use-urllib-request.patch
-    # Required because pastebin.com now redirects http requests to https
-    (fetchpatch {
-      name = "pastebin-com-https.patch";
-      url = "https://bazaar.launchpad.net/~arnouten/pastebinit/pastebin-com-https/diff/264?context=3";
-      sha256 = "0hxhhfcai0mll8qfyhdl3slmbf34ynb759b648x63274m9nd2kji";
-    })
+  buildInputs = [
+    (python3.withPackages (p: [ p.distro ]))
   ];
 
+  buildPhase = ''
+    xsltproc --nonet ${docbook-xsl-ns}/share/xml/docbook-xsl-ns/manpages/docbook.xsl pastebinit.xml
+  '';
+
   installPhase = ''
     mkdir -p $out/bin
     mkdir -p $out/etc
     cp -a pastebinit $out/bin
     cp -a pastebin.d $out/etc
-    substituteInPlace $out/bin/pastebinit --replace "'/etc/pastebin.d" "'$out/etc/pastebin.d"
+    substituteInPlace $out/bin/pastebinit --subst-var-by "etc" "$out/etc"
+    installManPage pastebinit.1
+  '';
+
+  doCheck = true;
+  nativeCheckInputs = buildInputs ++ [ wget cacert ];
+  checkPhase = ''
+    export USER="nixpkgs-pastebinit-tests"
+    teststring="test from pastebinit"
+    for pastebin in $(ls pastebin.d/*.conf | sed 's~^pastebin.d/\(.*\)\.conf$~\1~'); do
+      echo "Trying $pastebin"
+      URL=$(printf "$teststring\n$teststring\n$teststring" | python ./pastebinit -b $pastebin)
```

Comment:
A bit weird to publish stuff to the internet as part of the tests, which might be another reason to move it to `passthru.tests`

Reply:
I took inspiration from the upstream tests